crates/rpc/rpc-builder/src/lib.rs

```diff
@@ -529,10 +534,41 @@ impl RpcServerConfig {
     pub fn ipc(config: IpcServerBuilder) -> Self {
         Self::default().with_ipc(config)
     }
-
     /// Configures the http server
-    pub fn with_http(mut self, config: ServerBuilder) -> Self {
-        self.http_server_config = Some(config.http_only());
+    pub fn with_http(mut self, config: ServerBuilder, domains: Option<&String>) -> Self {
```

Comment:
The `domains` method should be separate, so one should be able to do `with_http(config).with_cors(Some("*"))`